PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/LaundryMachineAdapter.java

```diff
@@ -0,0 +1,192 @@
+                final CompoundButton button = buttonView;
+                if (isChecked) {
+                    final PendingIntent alarmIntent = PendingIntent.getBroadcast(getContext(), 50 * laundry.hall_no + machine.number,
+                            intent, PendingIntent.FLAG_CANCEL_CURRENT);
+                    alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP, SystemClock.elapsedRealtime() + 3000, alarmIntent);
+                    StringBuilder stringBuilder = new StringBuilder();
+                    stringBuilder.append(activity.getString(R.string.laundry_notification_snackbar_on))
+                            .append(" ").append(machine.machine_type).append(" ")
+                            .append(machine.number);
+                    Snackbar snackbar = Snackbar.make(buttonView,stringBuilder , Snackbar.LENGTH_SHORT)
+                            .setAction(activity.getString(R.string.laundry_undo), new View.OnClickListener(){
+                                @Override
+                                public void onClick(View view){
+                                    button.setChecked(false);
+                                    alarmManager.cancel(alarmIntent);
+                                    alarmIntent.cancel();
+                                }
+                            });
+                    View subView = snackbar.getView();
+                    TextView snackTextView = (TextView) subView.findViewById(android.support.design.R.id.snackbar_text);
+                    snackTextView.setTextColor(activity.getResources().getColor(R.color.white));
+                    snackbar.show();
+                } else {
+                    final PendingIntent alarmIntent = PendingIntent.getBroadcast(getContext(), 50 * laundry.hall_no + machine.number,
+                            intent, PendingIntent.FLAG_NO_CREATE);
+                    if (alarmIntent != null) {
+                        alarmManager.cancel(alarmIntent);
+                        alarmIntent.cancel();
+                    }
+                    StringBuilder stringBuilder = new StringBuilder();
+                    stringBuilder.append(activity.getString(R.string.laundry_notification_snackbar_off))
+                            .append(" ").append(machine.machine_type).append(" ")
+                            .append(machine.number);
+                    Snackbar snackbar = Snackbar.make(buttonView,stringBuilder, Snackbar.LENGTH_SHORT)
+                            .setAction(activity.getString(R.string.laundry_undo), new View.OnClickListener(){
+                                @Override
+                                public void onClick(View view){
+                                    button.setChecked(true);
+                                    PendingIntent alarmIntent = PendingIntent.getBroadcast(getContext(), 50 * laundry.hall_no + machine.number,
```

Comment:
Yes they should be defined as constants.  They are just placeholders so that I can test the code- 3000 corresponds to the milliseconds for the alarm (will be changed to 60*laundry.time_left <-- needs some parsing)
50 is a constant to be multiplied to the id so that every pending intent created by this function will not have overlapping ids.